src/imap/idle.rs

```diff
@@ -92,6 +96,9 @@ impl Session {
         session.as_mut().set_read_timeout(Some(IMAP_TIMEOUT));
         self.inner = session;
 
+        // Fetch mail once we exit IDLE.
```

Comment:
Do i correctly understand that this is just in case and not related to the fix itself? Can it be done under `if self.server_sent_unsolicited_exists(context)?` if so?